src/util/battery/batterylinux.cpp

```diff
@@ -7,99 +7,59 @@
 #include <QtDebug>
 
 BatteryLinux::BatteryLinux(QObject* pParent)
-    : Battery(pParent) {
+    : Battery(pParent),
+      m_client(up_client_new()) {
 }
 
 BatteryLinux::~BatteryLinux() {
+    g_object_unref(static_cast<UpClient*>(m_client));
 }
 
 void BatteryLinux::read() {
+    VERIFY_OR_DEBUG_ASSERT(static_cast<UpClient*>(m_client)) {
+        return;
+    }
     m_iMinutesLeft = Battery::TIME_UNKNOWN;
     m_dPercentage = 0.0;
     m_chargingState = Battery::UNKNOWN;
 
-    // NOTE(rryan): It would be nice if we could create the client
-    // once. However, while testing this up_client_get_devices(client) returned
-    // an empty list when I tried to re-use the UpClient instance.
-    UpClient* client = up_client_new();
-    VERIFY_OR_DEBUG_ASSERT(client) {
-      return;
+    guint kind = UP_DEVICE_KIND_UNKNOWN;
```

Comment:
Move the definition of those variables after the early exit when they are actually needed.

Why are some of them like `isPresent` initialized while others like `state` are not? Either initialize all or none. if `g_object_get()` guarantees to initialize them after successfully returning we should leave them uninitialized here.